### PR TITLE
En 4195 integration tests trie pruning

### PR DESCRIPTION
**Pull request by BeniaminDrasovean**

### CODE REVIEW

File: integrationTests/state/stateTrie_test.go [hunk 1]
```diff
@@ -1171,3 +1175,166 @@ func collapseTrie(state state.AccountHandler, t *testing.T) {
...
+	round := uint64(0)
+	nonce := uint64(0)
+
+	valMinting := big.NewInt(100)
+	valToTransferPerTx := big.NewInt(2)
+
+	advertiser := integrationTests.CreateMessengerWithKadDht(context.Background(), "")
+	_ = advertiser.Bootstrap()
+
+	nodes := integrationTests.CreateNodes(
```
Comment (iulianpascalau) [inline integrationTests/state/stateTrie_test.go]:
```GO
nodes, advertiser, idxProposers := integrationTests.SetupSyncNodesOneShardAndMeta(numNodesPerShard, numNodesMeta)
```

File: integrationTests/state/stateTrie_test.go [hunk 2]
```diff
@@ -1171,3 +1175,166 @@ func collapseTrie(state state.AccountHandler, t *testing.T) {
...
+		sendersPrivateKeys[i], _, _ = integrationTests.GenerateSkAndPkInShard(generateCoordinator, senderShard)
+		_, pk, _ := integrationTests.GenerateSkAndPkInShard(generateCoordinator, senderShard)
+		receiversPublicKeys[senderShard] = append(receiversPublicKeys[senderShard], pk)
+	}
+
+	fmt.Println("Minting sender addresses...")
+	integrationTests.CreateMintingForSenders(nodes, senderShard, sendersPrivateKeys, valMinting)
+
+	shardNode := nodes[0]
+	rootHash, _ := shardNode.AccntState.RootHash()
```
Comment (iulianpascalau) [inline integrationTests/state/stateTrie_test.go]: Also here, you take the root hash of the genesis block

File: integrationTests/testInitializer.go [hunk 3]
```diff
@@ -54,8 +54,14 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
-var stepDelay = time.Second
-var p2pBootstrapStepDelay = 5 * time.Second
+// StepDelay is used so that transactions can disseminate properly
+var StepDelay = time.Second
+
+// StepSync is used so that nodes have enough time to sync
+var StepSync = time.Second * 2
```
Comment (sasurobert) [inline integrationTests/testInitializer.go]: SyncDelay ?

File: integrationTests/testInitializer.go [hunk 4]
```diff
@@ -54,8 +54,14 @@ import (
...
+
+// P2pBootstrapStepDelay is used so that nodes have enough time to bootstrap
+var P2pBootstrapStepDelay = 5 * time.Second
```
Comment (sasurobert) [inline integrationTests/testInitializer.go]: P2PBootstrapDelay ?

File: integrationTests/state/stateTrie_test.go [hunk 5]
```diff
@@ -1171,3 +1175,166 @@ func collapseTrie(state state.AccountHandler, t *testing.T) {
...
+		sendersPrivateKeys[i], _, _ = integrationTests.GenerateSkAndPkInShard(generateCoordinator, 0)
+		_, pk, _ := integrationTests.GenerateSkAndPkInShard(generateCoordinator, 0)
+		receiversPublicKeys[0] = append(receiversPublicKeys[0], pk)
+	}
+
+	fmt.Println("Minting sender addresses...")
+	integrationTests.CreateMintingForSenders(nodes, 0, sendersPrivateKeys, valMinting)
+
+	shardNode := nodes[0]
+	rootHash, _ := shardNode.AccntState.RootHash()
```
[the remark this answers is not in the document]
Comment (BeniaminDrasovean) [reply integrationTests/state/stateTrie_test.go]: Right. Done.

File: integrationTests/state/stateTrie_test.go [hunk 6]
```diff
@@ -1171,3 +1175,166 @@ func collapseTrie(state state.AccountHandler, t *testing.T) {
...
+	integrationTests.ProposeBlocks(
+		nodes,
+		&round,
+		idxProposers,
+		nonces,
+		numOfRounds,
+	)
+
+	err := shardNode.AccntState.RecreateTrie(rootHash)
+	assert.Nil(t, err)
```
[the remark this answers is not in the document]
Comment (BeniaminDrasovean) [reply integrationTests/state/stateTrie_test.go]: Done.

File: integrationTests/state/stateTrie_test.go [hunk 7]
```diff
@@ -1171,3 +1175,166 @@ func collapseTrie(state state.AccountHandler, t *testing.T) {
...
+	integrationTests.EmptyDataPools(nodes, shardIdToRollbackLastBlock)
+
+	assert.Equal(t, uint64(1), nodes[0].BlockChain.GetCurrentBlockHeader().GetNonce())
+	assert.Equal(t, uint64(2), nodes[1].BlockChain.GetCurrentBlockHeader().GetNonce())
+
+	nonces := []*uint64{new(uint64), new(uint64)}
+	atomic.AddUint64(nonces[0], 2)
+	atomic.AddUint64(nonces[1], 3)
+
+	numOfRounds := 6
```
[the remark this answers is not in the document]
Comment (BeniaminDrasovean) [reply integrationTests/state/stateTrie_test.go]: Done.

File: integrationTests/state/stateTrie_test.go (hunk 1 above)
Comment (BeniaminDrasovean) [reply integrationTests/state/stateTrie_test.go]: Done.

File: integrationTests/state/stateTrie_test.go (hunk 2 above)
Comment (BeniaminDrasovean) [reply integrationTests/state/stateTrie_test.go]: Changed.

File: data/interface.go [hunk 8]
```diff
@@ -4,6 +4,12 @@ import (
 	"math/big"
 )
 
+// OldRootIdentifier is appended to the key when oldHashes are added to the evictionWaitingList
+var OldRootIdentifier = []byte("oldRoot")
```
[the remark this answers is not in the document]
Comment (BeniaminDrasovean) [reply data/interface.go]: Done.

**Pull request merged**